beacon-chain/rpc/eth/beacon/pool.go

```diff
@@ -363,6 +366,52 @@ func (bs *Server) SubmitSignedBLSToExecutionChanges(ctx context.Context, req *et
 	return &emptypb.Empty{}, nil
 }
 
+func (bs *Server) broadcastBLSBatch(ctx context.Context, ptr *[]*ethpbalpha.SignedBLSToExecutionChange) {
+	changes := *ptr
+	limit := broadcastBLSChangesRateLimit
+	if len(changes) < broadcastBLSChangesRateLimit {
+		limit = len(changes)
+	}
+	st, err := bs.ChainInfoFetcher.HeadState(ctx)
```

Comment:
We could still have a lot of goroutines spawning that call this method and trigger a call to HeadState unless we move to a background receive, aggregate, batch broadcast approach